kernel/src/hil/digest.rs

```diff
@@ -0,0 +1,72 @@
+//! Interface for Digest
+
+use crate::common::leasable_buffer::LeasableBuffer;
+use crate::returncode::ReturnCode;
+
+/// The 'types' of digests, this should define the output size of the digest
+/// operations.
+pub trait DigestType: Eq + Copy + Clone + Sized + AsRef<[u8]> + AsMut<[u8]> {}
+
+impl DigestType for [u8; 32] {}
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a, T: DigestType> {
+    /// This callback is called when the data has been added to the digest
+    /// engine.
+    /// On error or success `data` will contain a reference to the original
+    /// data supplied to `add_data()`.
+    fn add_data_done(&'a self, result: Result<(), ReturnCode>, data: &'static mut [u8]);
```

Comment:
If we're already forcing the client to use a `LeasableBuffer` in the `add_data()` method, wouldn't it be reasonable to return the same type here as well?

Reply:
I don't think so as the client doesn't really want to use a `LeasableBuffer`. The  `LeasableBuffer` is just equivalent to passing in the length.